compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -273,7 +273,7 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
     val cls = mirroredType.classSymbol
     val useCompanion = cls.useCompanionAsSumMirror
 
-    if cls.isGenericSum(if useCompanion then cls.linkedClass else ctx.owner) then
+    if (!mirroredType.termSymbol.isEnumCase && (cls.isGenericSum(if useCompanion then cls.linkedClass else ctx.owner))) then
```

Comment:
In SyntheticMembers::addMirrorSupport? It's not an issue, because this method will never be called: as far as I understand branch `else if (impl.removeAttachment(ExtendsSingletonMirror).isDefined)` will be invoked.

Reply:
it's fine to keep `isGenericSum` implementation unchanged, perhaps it could be more informative to rename to `isGenericSumClass, along with `isGenericProduct` to `isGenericProductClass`